doc/design/requirements.md

```diff
@@ -0,0 +1,69 @@
+# Requirements and goals
+
+Includes the overlapping projects of TiUp, tiup-cluster, and tiup-playground.
+
+We should distinguish between goals and requirements for the long term vs an initial release.
+
+## Vision and high-level goals
+
+* Manage and deploy TiDB and TiKV clusters
+* Open source project, with a community of users and contributors
+* Help TiDB and TiKV developers run and test their code in a production-like environment
+* Make TiDB an easy to use database
+* Be the standard way for TiDB users to manage their clusters
+* Make it easy to try out TiDB, both on a single machine and a cluster
+* Extensible
+
+## Requirements
```

Comment:
I think the requirements can be expand the following core ideas:
1. We hope users can try it out as soon as they know TiDB.
2. Some developers can set up a local debug environment to adapt their application.
3. We can quickly set up a cluster in any version to reproduce the BUG reported by the ONCALL or GitHub issue.
4. The users can provide a reproduce easily via `tiup`, ref: https://github.com/pingcap/tidb/issues/737.
5. We should provide a component to make operate production clusters easier.
6. We should output a more human-friendly error message when some unexpected occur.

Reply:
"We should output a more human-friendly error message when some unexpected occur." - could you give an example of this please?

Reply:
The initial idea we want to re-implement the `tiup-cluster` is that the ansible cannot emit a human-friendly error message, and the users cannot figure out the root cause of why the procedure abort.

@breeswish introduces a new error message framework (https://github.com/pingcap-incubator/tiup-cluster/blob/master/cmd/cluster/command/root.go#L108 /  https://github.com/pingcap-incubator/tiup-cluster/pull/116). We can attach more details and clear error message context, and give some fixed steps to help the users learn the reason for errors and how to fix them.